docs/secret_management.md

```diff
@@ -192,68 +164,55 @@ metadata:
-          secret:
-            secretName: test-secret
-    clusterAgent:
-      containers:
-        cluster-agent:
-          env:
-            - name: DD_SECRET_BACKEND_COMMAND
-              value: "/readsecret.sh"
-            - name: DD_SECRET_BACKEND_ARGUMENTS
-              value: "/etc/secret-volume"
-          volumeMounts:
-            - name: secret-volume
-              mountPath: "/etc/secret-volume"
-      volumes:
-        - name: secret-volume
-          secret:
-            secretName: test-secret
-```
-
-The Datadog Agent also includes a script that can be used to read secrets from files mounted from Kubernetes secrets, or directly from Kubernetes secrets. This script can be used by setting `DD_SECRET_BACKEND_COMMAND` to `/readsecret_multiple_providers.sh`. An example of how to configure the DatadogAgent spec is provided below. For more details, see [Secrets Management][2].
-
-```yaml
-apiVersion: datadoghq.com/v2alpha1
-kind: DatadogAgent
-metadata:
-  name: datadog
-spec:
-  global:
+    secretBackend:
+      command: "/readsecret_multiple_providers.sh"
     credentials:
       apiKey: ENC[k8s_secret@default/test-secret/api_key]
       appKey: ENC[k8s_secret@default/test-secret/app_key]
-  override:
-    nodeAgent:
-      env:
-        - name: DD_SECRET_BACKEND_COMMAND
-          value: "/readsecret_multiple_providers.sh"
 ```
 
 **Remarks:**
 
-* For the "Agent" and "Cluster Agent", others options exist to configure secret backend command:
+* The `"/readsecret_multiple_providers.sh"` helper enables the Agent to directly read Kubernetes Secrets across both its own and other namespaces. Ensure that the associated ServiceAccount has the necessary permissions by assigning the appropriate Roles and RoleBindings, which can be set manually or using the following options:
+    * `global.secretBackend.enableGlobalPermissions`: Determines if a ClusterRole is created that enables the Agents to read **all** Kubernetes Secrets.
+      ```yaml
+      apiVersion: datadoghq.com/v2alpha1
+      kind: DatadogAgent
+      metadata:
+        name: datadog
+      spec:
+        global:
+          secretBackend:
+            command: "/readsecret_multiple_providers.sh"
+            enableGlobalPermissions: true
+      # ...
+      ```
+    * `global.secretBackend.roles`: Replaces `enableGlobalPermissions`, detailing the list of namespaces/secrets to which the Agents should have access.
+      ```yaml
+      apiVersion: datadoghq.com/v2alpha1
+      kind: DatadogAgent
+      metadata:
+        name: datadog
+      spec:
+        global:
+          secretBackend:
+            command: "/readsecret_multiple_providers.sh"
+            roles:
+            - namespace: rabbitmq-system
+              secrets:
+              - "rabbitmqcluster-sample-default-user"
+      # ...
+      ```
+      In this example, a Role is created granting read access to the secret `rabbitmqcluster-sample-default-user` in the `rabbitmq-system` namespace.
```

Comment:
I'm being nitpicky here, but I notice a couple places we capitalize "secret" as a proper noun (as in the kubernetes docs https://kubernetes.io/docs/concepts/configuration/secret/) and some places we don't..

Reply:
Adressed in 20a4b0993fd214ecf5c54bd9fa418b623f7d88ef. Only left `namespace/secrets` as they reference the CRD keys